Needs update for numpy 2

Is this package still maintained? It would need an update for numpy2: 

```python
[  171s] ============================= test session starts ==============================
[  171s] platform linux -- Python 3.10.15, pytest-8.3.2, pluggy-1.5.0 -- /usr/bin/python3.10
[  171s] cachedir: .pytest_cache
[  171s] rootdir: /home/abuild/rpmbuild/BUILD/GooseMPL-0.15.0
[  171s] configfile: pyproject.toml
[  171s] collecting ... collected 33 items
[  171s]
[  171s] tests/test_main.py::Test_ticks::test_log_minorticks PASSED               [  3%]
[  171s] tests/test_main.py::Test_ticks::test_log_minorticks_plot PASSED          [  6%]
[  171s] tests/test_main.py::Test_ticks::test_log_ticks PASSED                    [  9%]
[  171s] tests/test_main.py::Test_ticks::test_log_ticks_plot PASSED               [ 12%]
[  171s] tests/test_main.py::Test_fit_powerlaw::test_exponent PASSED              [ 15%]
[  171s] tests/test_main.py::Test_fit_powerlaw::test_prefactor PASSED             [ 18%]
[  171s] tests/test_main.py::Test_fit_powerlaw::test_prefactor_exponent PASSED    [ 21%]
[  171s] tests/test_main.py::Test_fit_exp::test_exponent PASSED                   [ 24%]
[  171s] tests/test_main.py::Test_fit_exp::test_prefactor PASSED                  [ 27%]
[  171s] tests/test_main.py::Test_fit_exp::test_prefactor_exponent PASSED         [ 30%]
[  171s] tests/test_main.py::Test_fit_exp::test_prefactor_negative_exponent PASSED [ 33%]
[  171s] tests/test_main.py::Test_fit_log::test_exponent PASSED                   [ 36%]
[  171s] tests/test_main.py::Test_fit_log::test_prefactor PASSED                  [ 39%]
[  171s] tests/test_main.py::Test_fit_log::test_prefactor_exponent PASSED         [ 42%]
[  171s] tests/test_main.py::Test_fit_log::test_prefactor_negative_prefactor PASSED [ 45%]
[  171s] tests/test_main.py::Test_fit_linear::test_offset PASSED                  [ 48%]
[  171s] tests/test_main.py::Test_fit_linear::test_offset_slope PASSED            [ 51%]
[  171s] tests/test_main.py::Test_fit_linear::test_slope PASSED                   [ 54%]
[  171s] tests/test_main.py::Test_cdf::test_random PASSED                         [ 57%]
[  171s] tests/test_main.py::Test_cdf::test_simple PASSED                         [ 60%]
[  171s] tests/test_main.py::Test_bin::test_simple FAILED                         [ 63%]
[  171s] tests/test_main.py::Test_histogram_norm::test_density PASSED             [ 66%]
[  171s] tests/test_main.py::Test_histogram_rebin::test_a PASSED                  [ 69%]
[  171s] tests/test_main.py::Test_histogram_rebin::test_a_strip PASSED            [ 72%]
[  171s] tests/test_main.py::Test_histogram_rebin::test_b PASSED                  [ 75%]
[  171s] tests/test_main.py::Test_histogram_rebin::test_c PASSED                  [ 78%]
[  171s] tests/test_main.py::Test_histogram_rebin::test_d PASSED                  [ 81%]
[  171s] tests/test_main.py::Test_histogram_bin_edges2midpoint::test_simple PASSED [ 84%]
[  171s] tests/test_main.py::Test_histogram_bin_edges_integer::test_back PASSED   [ 87%]
[  171s] tests/test_main.py::Test_histogram_bin_edges_integer::test_front PASSED  [ 90%]
[  171s] tests/test_main.py::Test_histogram_bin_edges_integer::test_middle PASSED [ 93%]
[  171s] tests/test_main.py::Test_histogram_bin_edges::test_integer PASSED        [ 96%]
[  171s] tests/test_main.py::Test_histogram_bin_edges::test_integer2 PASSED       [100%]
[  171s]
[  171s] =================================== FAILURES ===================================
[  171s] _____________________________ Test_bin.test_simple _____________________________
[  171s]
[  171s] self = <tests.test_main.Test_bin testMethod=test_simple>
[  171s]
[  171s]     def test_simple(self):
[  171s]         xdata = np.array([1, 1, 3, 3, 3, 5, 5])
[  171s]         ydata = np.array([2, 4, 1, 2, 3, 2, 4])
[  171s]         bin_edges = np.array([0, 2, 4, 6])
[  171s]
[  171s] >       data = gplt.bin(xdata, ydata, bin_edges)
[  171s]
[  171s] tests/test_main.py:307:
[  171s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[  171s] GooseMPL/__init__.py:2416: in bin
[  171s]     "x": np.NaN * np.ones(n, dtype=float),
[  171s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[  171s]
[  171s] attr = 'NaN'
[  171s]
[  171s]     def __getattr__(attr):
[  171s]         # Warn for expired attributes
[  171s]         import warnings
[  171s]
[  171s]         if attr == "linalg":
[  171s]             import numpy.linalg as linalg
[  171s]             return linalg
[  171s]         elif attr == "fft":
[  171s]             import numpy.fft as fft
[  171s]             return fft
[  171s]         elif attr == "dtypes":
[  171s]             import numpy.dtypes as dtypes
[  171s]             return dtypes
[  171s]         elif attr == "random":
[  171s]             import numpy.random as random
[  171s]             return random
[  171s]         elif attr == "polynomial":
[  171s]             import numpy.polynomial as polynomial
[  171s]             return polynomial
[  171s]         elif attr == "ma":
[  171s]             import numpy.ma as ma
[  171s]             return ma
[  171s]         elif attr == "ctypeslib":
[  171s]             import numpy.ctypeslib as ctypeslib
[  171s]             return ctypeslib
[  171s]         elif attr == "exceptions":
[  171s]             import numpy.exceptions as exceptions
[  171s]             return exceptions
[  171s]         elif attr == "testing":
[  171s]             import numpy.testing as testing
[  171s]             return testing
[  171s]         elif attr == "matlib":
[  171s]             import numpy.matlib as matlib
[  171s]             return matlib
[  171s]         elif attr == "f2py":
[  171s]             import numpy.f2py as f2py
[  171s]             return f2py
[  171s]         elif attr == "typing":
[  171s]             import numpy.typing as typing
[  171s]             return typing
[  171s]         elif attr == "rec":
[  171s]             import numpy.rec as rec
[  171s]             return rec
[  171s]         elif attr == "char":
[  171s]             import numpy.char as char
[  171s]             return char
[  171s]         elif attr == "array_api":
[  171s]             raise AttributeError("`numpy.array_api` is not available from "
[  171s]                                  "numpy 2.0 onwards", name=None)
[  171s]         elif attr == "core":
[  171s]             import numpy.core as core
[  171s]             return core
[  171s]         elif attr == "strings":
[  171s]             import numpy.strings as strings
[  171s]             return strings
[  171s]         elif attr == "distutils":
[  171s]             if 'distutils' in __numpy_submodules__:
[  171s]                 import numpy.distutils as distutils
[  171s]                 return distutils
[  171s]             else:
[  171s]                 raise AttributeError("`numpy.distutils` is not available from "
[  171s]                                      "Python 3.12 onwards", name=None)
[  171s]
[  171s]         if attr in __future_scalars__:
[  171s]             # And future warnings for those that will change, but also give
[  171s]             # the AttributeError
[  171s]             warnings.warn(
[  171s]                 f"In the future `np.{attr}` will be defined as the "
[  171s]                 "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
[  171s]
[  171s]         if attr in __former_attrs__:
[  171s]             raise AttributeError(__former_attrs__[attr], name=None)
[  171s]
[  171s]         if attr in __expired_attributes__:
[  171s] >           raise AttributeError(
[  171s]                 f"`np.{attr}` was removed in the NumPy 2.0 release. "
[  171s]                 f"{__expired_attributes__[attr]}",
[  171s]                 name=None
[  171s]             )
[  171s] E           AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.
[  171s]
[  171s] /usr/lib64/python3.10/site-packages/numpy/__init__.py:400: AttributeError
[  171s] =========================== short test summary info ============================
[  171s] FAILED tests/test_main.py::Test_bin::test_simple - AttributeError: `np.NaN` w...
[  171s] ========================= 1 failed, 32 passed in 0.55s =========================
```

--- a/GooseMPL/__init__.py
+++ b/GooseMPL/__init__.py
@@ -2413,10 +2413,10 @@ def bin(
     n = bin_edges.size - 1
 
     ret = {
-        "x": np.NaN * np.ones(n, dtype=float),
-        "y": np.NaN * np.ones(n, dtype=float),
-        "xerr": np.NaN * np.ones(n, dtype=float),
-        "yerr": np.NaN * np.ones(n, dtype=float),
+        "x": np.nan * np.ones(n, dtype=float),
+        "y": np.nan * np.ones(n, dtype=float),
+        "xerr": np.nan * np.ones(n, dtype=float),
+        "yerr": np.nan * np.ones(n, dtype=float),
         "n": np.zeros(n, dtype=int),
     }
 

